File: 145-binary-tree-postorder-traversal/binary-tree-postorder-traversal.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
void traverse(struct TreeNode* node, int** res, int* returnSize) {
    if (node == NULL) {
        return;
    }
    traverse(node->left, res, returnSize);
    traverse(node->right, res, returnSize);
    (*res)[(*returnSize)++] = node->val;
}

int* postorderTraversal(struct TreeNode* root, int* returnSize) {
    int* res = (int*)malloc(100 * sizeof(int));
    *returnSize = 0;
    traverse(root, &res, returnSize);
    return res;
}

File: 145-binary-tree-postorder-traversal/binary-tree-postorder-traversal.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def postorderTraversal(self, root):
        res = []
        def traverse(node):
            if not node:
                return
            traverse(node.left)
            traverse(node.right)
            res.append(node.val)
        traverse(root)
        return res
